Sources/XCTest/WallClockTimeMetric.swift

```diff
@@ -0,0 +1,86 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//
+//  WallClockTimeMetric.swift
+//  Performance metric measuring how long it takes code to execute
+//
+
+#if os(Linux) || os(FreeBSD)
+    import Foundation
+#else
+    import SwiftFoundation
+#endif
+
+/// This metric uses the system uptime to keep track of how much time passes
+/// between starting and stopping measuring.
+internal final class WallClockTimeMetric: PerformanceMetric {
+    static let name = "org.swift.XCTPerformanceMetric_WallClockTime"
+
+    typealias Measurement = NSTimeInterval
+    private var startTime: NSTimeInterval?
+    var measurements: [Measurement] = []
+
+    func startMeasuring() {
+        startTime = currentTime()
+    }
+
+    func stopMeasuring() {
+        guard let startTime = startTime else { fatalError("Must start measuring before stopping measuring") }
+        let stopTime = currentTime()
+        measurements.append(stopTime-startTime)
+    }
+
+    private let maxRelativeStandardDeviation = 10.0
+    private let standardDeviationNegligibilityThreshold = 0.1
+
+    func calculateResults() -> String {
+        let results = [
+                          String(format: "average: %.3f", measurements.average),
+                          String(format: "relative standard deviation: %.3f%%", measurements.relativeStandardDeviation),
+                          "values: [\(measurements.map({ String(format: "%.6f", $0) }).joined(separator: ", "))]",
+                          "performanceMetricID:\(self.dynamicType.name)",
+                          String(format: "maxPercentRelativeStandardDeviation: %.3f%%", maxRelativeStandardDeviation),
+                          String(format: "maxStandardDeviation: %.3f", standardDeviationNegligibilityThreshold),
+                          ]
+        return "[Time, seconds] \(results.joined(separator: ", "))"
+    }
+
+    func failureMessage() -> String? {
+        // The relative standard deviation of the measurements is \d+.\d{3}% which is higher than the max allowed of \d+.\d{3}%.
```

Comment:
Ooooops, that line wan't supposed to be included here, good catch. 😆 I had copied it in from the test file for reference when I was writing the method.